src/admin/render_readmes.rs

```diff
@@ -290,6 +294,37 @@ readme = "README.md"
         assert!(result.contains("readme"))
     }
 
+    #[test]
+    fn test_render_pkg_no_readme() {
+        let mut pkg = tar::Builder::new(vec![]);
+        add_file(
+            &mut pkg,
+            "foo-0.0.1/Cargo.toml",
+            br#"
+[package]
+"#,
+        );
+        let serialized_archive = pkg.into_inner().unwrap();
+        assert!(render_pkg_readme(tar::Archive::new(&*serialized_archive), "foo-0.0.1").is_none());
+    }
+
+    #[test]
+    fn test_render_pkg_implicit_readme() {
+        let mut pkg = tar::Builder::new(vec![]);
+        add_file(
+            &mut pkg,
+            "foo-0.0.1/Cargo.toml",
+            br#"
+[package]
+"#,
+        );
+        add_file(&mut pkg, "foo-0.0.1/README.md", b"readme");
+        let serialized_archive = pkg.into_inner().unwrap();
+        let result =
+            render_pkg_readme(tar::Archive::new(&*serialized_archive), "foo-0.0.1").unwrap();
+        assert!(result.contains("readme"))
```

Comment:
any reason for using `contains` here instead of `assert_eq`?

Reply:
For the shorter ones like this, I could do assert_eq as it comes out to
```<p>readme</p>\n```

For the longer ones, they're really verbose and full of stuff not relevant to the test itself, so I was using the `result.contains` pattern instead.
```
<p>docs/readme <a href=\"https://github.com/foo/foo/blob/HEAD/docs/./Other.md\" rel=\"nofollow noopener noreferrer\">link</a></p>\n
```

What do you think? Switch them over?

Reply:
I would have still used `assert_eq`, but I guess it's fine like this too :)